File: 1601-1700/1688-count-of-matches-in-tournament.go
package main

import "fmt"

func numberOfMatches(n int) int {
    cnt := 0
    for n > 1 {
        if n % 2 == 0 {
            cnt += n / 2
            n /= 2
        } else {
            cnt += (n - 1) / 2
            n = (n - 1) / 2 + 1
        }
    }
    return cnt
}

func main() {
    fmt.Println(numberOfMatches(7));
}

File: 1601-1700/1688-count-of-matches-in-tournament.cpp
class Solution {
public:
    int numberOfMatches(int n) {
        int ans = 0;
        while (n > 1) {
            ans += n / 2;
            if (n & 1) {
                n = n / 2 + 1;
            } else n /= 2;
        }
        return ans;
    }
};
